src/abstract-interpretation/data-frame/mappers/arguments.ts

```diff
@@ -0,0 +1,270 @@
+ */
+export interface FunctionParameterLocation<T = undefined> {
+    pos:      number,
+    name?:    string
+    default?: T
+}
+
+/**
+ * Escapes a regular expression given as string by escaping all special regular expression characters.
+ *
+ * @param text        - The text to escape
+ * @param allowTokens - Whether to allow and keep unescaped tokens like `\s`, `\t`, or `\n`
+ * @returns The escaped text
+ */
+export function escapeRegExp(text: string, allowTokens: boolean = false): string {
+	if(allowTokens) {  // only allow and keep the tokens `\s`, `\t`, and `\n` in the text
+		return text.replace(/[.*+?^${}()|[\]]|\\[^stn]/g, '\\$&');
+	} else {
+		return text.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+	}
+}
+
+/**
+ * Maps all invalid, duplicate, or empty column names to top depending on the provided arguments.
+ *
+ * @param colnames     - The columns names to filter
+ * @param checkNames   - Whether to map all invalid column names to top (`undefined`)
+ * @param noDupNames   - Whether to map all duplicate column names to top (`undefined`)
+ * @param noEmptyNames - Whether to map all empty column names to top (`undefined`)
+ * @returns The filtered column names
+ */
+export function filterValidNames(
+	colnames: (string | undefined)[] | undefined,
+	checkNames?: boolean,
+	noDupNames?: boolean,
+	noEmptyNames?: boolean
+): (string | undefined)[] | undefined {
+	if(checkNames) {  // map all invalid column names to top
+		colnames = colnames?.map(entry => isValidColName(entry) ? entry : undefined);
+	}
+	if(noDupNames) {  // map all duplicate column names to top
+		colnames = colnames?.map((entry, _, list) => entry !== undefined && list.filter(other => other === entry).length === 1 ? entry : undefined);
+	}
+	if(noEmptyNames) {  // map all empty column names to top
+		colnames = colnames?.map(entry => entry?.length === 0 ? undefined : entry);
+	}
+	return colnames;
+}
+
+/**
+ * Gets the value of an argument that specified as {@link FunctionParameterLocation}.
+ *
+ * @param args     - The arguments to get the requested argument from
+ * @param argument - The specification of the argument to get the value for
+ * @param info     - Argument resolve information
+ * @returns The resolved value of the argument or `undefined`
+ */
+export function getArgumentValue<T>(
+	args: readonly RFunctionArgument<ParentInformation>[],
+	argument: FunctionParameterLocation<T> | string,
+	info: ResolveInfo
+) {
+	const arg = getFunctionArgument(args, argument, info);
+	const defaultValue = typeof argument !== 'string' ? argument.default : undefined;
+
+	return arg !== undefined ? resolveIdToArgValue(arg, info) : defaultValue;
+}
+
+/**
+ * Gets all effective argument from a list of arguments by removing all arguments whose names should be excluded.
+ *
+ * @param args     - The list of arguments to filter
+ * @param excluded - The names of the arguments to exclude
+ * @returns The filtered list of arguments
+ */
+export function getEffectiveArgs(
```

Comment:
given that this name is also used at a another position where it does additionally consider special operators, I would prefer one of those being renamed.